src/snowcli/cli/nativeapp/version/version_processor.py

```diff
@@ -0,0 +1,312 @@
+import logging
+from pathlib import Path
+from sys import stdin, stdout
+from textwrap import dedent
+from typing import Dict, List, Optional
+
+import typer
+from click import ClickException
+from git import Repo
+from git.exc import InvalidGitRepositoryError
+from rich import print
+from snowcli.cli.nativeapp.artifacts import find_version_info_in_manifest_file
+from snowcli.cli.nativeapp.constants import VERSION_COL
+from snowcli.cli.nativeapp.exceptions import ApplicationPackageDoesNotExistError
+from snowcli.cli.nativeapp.manager import (
+    NativeAppCommandProcessor,
+    NativeAppManager,
+    ensure_correct_owner,
+)
+from snowcli.cli.nativeapp.policy import PolicyBase, TaskContext
+from snowcli.cli.nativeapp.run_processor import NativeAppRunProcessor
+from snowcli.cli.nativeapp.utils import (
+    find_all_rows,
+    find_first_row,
+    is_user_in_interactive_mode,
+)
+from snowcli.cli.project.util import unquote_identifier
+from snowcli.exception import SnowflakeSQLExecutionError
+from snowflake.connector import ProgrammingError
+from snowflake.connector.cursor import DictCursor
+
+# Custom lambdas to plug in to UserConfirmationPolicy
+is_force_lambda = lambda force: force
+is_interactive_mode_lambda = lambda: stdin.isatty() and stdout.isatty()
+confirm_with_user_lambda = lambda prompt: typer.confirm(prompt)
+
+log = logging.getLogger(__name__)
+
+
+def check_index_changes_in_git_repo(
+    project_root: Path,
+    policy: PolicyBase,
+) -> None:
+    """
+    Checks if the project root, i.e. the native apps project is a git repository. If it is a git repository,
+    it also checks if there any local changes to the directory that may not be on the app package stage.
+    """
+    try:
+        repo = Repo(project_root)
+
+        # Check if the repo has any changes, including untracked files
+        if repo.is_dirty(untracked_files=True):
+            print("Changes detected in your git repository!")
+            # show differences between current files and last commit
+            print(repo.git.diff(repo.head.commit.tree))
```

Comment:
This is a bit odd to me for a few reasons:
-  git status would be better at showing the diff, no? What does this look like for the user?
- the only way out of this is --force, which is rather dangerous. Should we have a git-specific flag to disable this check? Something like --ignore-git or similar.
- still somewhat worried about the fact that the project root might not be the git root. If we had a "no git" flag, maybe that's a moot point.

Reply:
Answering both above comments in this one:
1. On whether we should check for git repositories (dirty or not) - this is based off of the functional specs. We can always revisit, but as discussed offline, let's discuss on a different channel.
2. I am neutral about git diff vs git status, will change it in the next commit. 
3. This git check is not behind a `--force` check. It happens regardless the policy.
4. project root not being the git root is taken care of by `search_parent_directories=True` coming up in the next commit. If the cwd is in some child directory (or further down), it will still be able to detect if it is a git repo.

Reply:
See my comments in the func spec